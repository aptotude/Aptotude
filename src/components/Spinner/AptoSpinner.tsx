import React from 'react';
import classNames from 'classnames';
import './aptoSpinner.scss';
import { StandardTypes } from '../../utils/standardTypes';

interface Props extends StandardTypes {
  text?: string;
  noOverlay?: boolean;
}

const COMPONENT_PREFIX = 'AptoSpinner';

export class AptoSpinner extends React.Component<Props> {
  public static defaultProps = {
    text: '',
    noOverlay: false
  };

  public render() {
    const {
      text,
      noOverlay,
      className,
      forwardRef,
      ...rest
    } = this.props;

    const classes = classNames(
      COMPONENT_PREFIX,
      noOverlay && `${COMPONENT_PREFIX}--noOverlay`,
      className
    );

    return (
      <div
        {...rest}
        ref={forwardRef}
        className={classes}>
          <div className="AptoSpinner-spinner">
            <svg viewBox="0 0 64 64">
              <circle transform="translate(32,32)" r="26"/>
            </svg>
          </div>
          {
            text ? <div className="AptoSpinner-content">{text}</div> : null
          }
      </div>
    );
  }
}
