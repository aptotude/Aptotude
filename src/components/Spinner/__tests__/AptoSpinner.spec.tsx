import React from 'react';
import renderer from 'react-test-renderer';
import AptoSpinner from '../AptoSpinner';

test('Spinner renders', () => {
  const component = renderer.create(
    <AptoSpinner />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Spinner with no overlay renders', () => {
  const component = renderer.create(
    <AptoSpinner noOverlay />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Spinner with text renders', () => {
  const component = renderer.create(
    <AptoSpinner text="foo" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Spinner with text and no overlay renders', () => {
  const component = renderer.create(
    <AptoSpinner text="foo" noOverlay />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
