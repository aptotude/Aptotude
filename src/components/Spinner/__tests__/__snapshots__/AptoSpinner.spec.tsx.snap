// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders 1`] = `
<div
  className="AptoSpinner"
>
  <div
    className="AptoSpinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner with no overlay renders 1`] = `
<div
  className="AptoSpinner AptoSpinner--noOverlay"
>
  <div
    className="AptoSpinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner with text and no overlay renders 1`] = `
<div
  className="AptoSpinner AptoSpinner--noOverlay"
>
  <div
    className="AptoSpinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
  <div
    className="AptoSpinner-content"
  >
    foo
  </div>
</div>
`;

exports[`Spinner with text renders 1`] = `
<div
  className="AptoSpinner"
>
  <div
    className="AptoSpinner-spinner"
  >
    <svg
      viewBox="0 0 64 64"
    >
      <circle
        r="26"
        transform="translate(32,32)"
      />
    </svg>
  </div>
  <div
    className="AptoSpinner-content"
  >
    foo
  </div>
</div>
`;
